pkg/stream/client.go

```diff
@@ -617,6 +617,12 @@ func (c *Client) BrokerForConsumer(stream string) (*Broker, error) {
 	if streamMetadata.responseCode != responseCodeOk {
 		return nil, lookErrorCode(streamMetadata.responseCode)
 	}
+
+	if streamMetadata.Leader == nil {
+		// in this case the leader is not ready yet
+		return nil, fmt.Errorf("leader not ready yet for stream: %s", stream)
+	}
+
 	var brokers []*Broker
 	brokers = append(brokers, streamMetadata.Leader)
 	brokers = append(brokers, streamMetadata.Replicas...)
```

Comment:
Is there a possibility for the `Replicas` array to contain `nil` brokers as well? If so, maybe we should check those too (if not to err, to dispose of the nil referencese)

Reply:
Also related to that, is there any possibility that the leader is not ready but the replicas are, and that we could still create a connection to start consuming from it (just not producing)? If so, it might be worth some check here for that case. 

Though we might want to only connect to the stream if the leader is really there anyway, avoiding more complex logics here for not so common/useful corner cases.

Reply:
Added a check also for the replicas